primitives/keystore/src/lib.rs

```diff
@@ -0,0 +1,386 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2019-2020 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Keystore traits
+pub mod testing;
+#[cfg(feature = "std")]
+pub mod vrf;
+
+use std::sync::Arc;
+use async_std::task::block_on;
```

Comment:
```suggestion
// Use `async_std::task::block_on` instead of `futures::executor::block_on` given that the former allows nested `block_on` calls without panics (see https://github.com/rust-lang/futures-rs/issues/2090 for details). By using `async_std::task::block_on` in `SyncCryptoStore` one can `futures::executor::block_on` on a future calling methods on `SyncCryptoStore`. This is e.g. used in unit testing.
use async_std::task::block_on;
```

Would you mind adding something along these lines for future us? (With line wrapping.)